test/testsets/GlassCat.jl

```diff
@@ -2,216 +2,343 @@
+            C4 = [2.277823700E+00, 3.239318260E-01, 1.997453950E-03, 2.445302672E-04],
+            C5 = [2.126055280E+00, 6.863473749E-01, 4.975938104E-01, 6.352738622E-01],
+            C6 = [3.227490100E+03, 1.625888344E+03, 1.443307391E+03, 1.397305161E+03],
+            C7 = repeat([0.0], 4),
+            C8 = repeat([0.0], 4),
+            C9 = repeat([0.0], 4),
+            C10 = repeat([0.0], 4),
+            D₀ = [2.226000000E-05, 1.122401430E-04, -1.718689957E-05, -2.545621409E-07],
+            D₁ = [5.150100000E-08, 2.868995096E-13, -6.499841076E-13, 1.258087492E-10],
+            D₂ = [1.309700000E-10, -4.157332734E-16, -4.153311088E-16, 1.119703940E-11],
+            E₀ = [4.382400000E-05, 9.532969159E-05, 8.220254442E-06, 2.184273853E-05],
+            E₁ = [4.966000000E-07, -1.045247704E-11, 6.086168578E-12, -2.788674395E-09],
+            λₜₖ = [3.728000000E-01, 6.338475915E-01, 3.610058073E-01, -1.948154058E+00],
+            temp = [2.002000000E+01, 2.000000000E+01, 2.000000000E+01, 2.000000000E+01],
+            relcost = [1.1, -1.0, -1.0, -1.0],
+            CR = [1.2, -1.0, -1.0, -1.0],
+            FR = [1.3, -1.0, -1.0, -1.0],
+            SR = [1.4, -1.0, -1.0, -1.0],
+            AR = [1.5, -1.0, -1.0, -1.0],
+            PR = [1.6, -1.0, -1.0, -1.0],
+            λmin = repeat([2.0], 4),
+            λmax = repeat([14.0], 4),
+            transmission = [
+                [[2.0, 1.0, 3.0], [14.0, 4.0, 5.0]],
+                [[2.0, 1.0, 1.0], [14.0, 1.0, 1.0]],
+                [[2.0, 1.0, 1.0], [14.0, 1.0, 1.0]],
+                [[2.0, 1.0, 1.0], [14.0, 1.0, 1.0]],
+            ],
+        )
+        FIELDS = names(TEST_CAT_VALUES)[2:end]
+
+        @testset "sourcefile_to_catalog" begin
+            catalog = GlassCat.sourcefile_to_catalog(SOURCE_FILE)
+
+            for row in eachrow(TEST_CAT_VALUES)
+                name = row["name"]
+                @test name ∈ keys(catalog)
+                for field in FIELDS
+                    @test row[field] == catalog[name][field]
+                end
+            end
+        end
 
-    @testset "Build Tests" begin
-        # check that all automatic downloads are working
-        for catname in split("HOYA NIKON OHARA SCHOTT Sumita")
-            agffile = joinpath(AGF_DIR, catname * ".agf")
-            @test isfile(agffile)
+        # integration test
+        @testset "generate_jls" begin
+            jldir = mktempdir()
+            mainfile = joinpath(jldir, "AGF_TEST_CAT.jl")
+
+            GlassCat.generate_jls([CATALOG_NAME], mainfile, jldir, SOURCE_DIR)
+            include(mainfile)
+
+            for row in eachrow(TEST_CAT_VALUES)
+                name = row["name"]
+                glass = getfield(getfield(Main, Symbol(CATALOG_NAME)), Symbol(name))
+                @test "$CATALOG_NAME.$name" ∈ getfield(Main, Symbol("AGF_GLASS_NAMES"))
+                @test_broken Symbol("$CATALOG_NAME.$name") ∈ getfield(Main, Symbol("AGF_GLASSES")) # ! TODO !
```

Comment:
It shouldn't be - it's just another issue that I've procrastinated, admittedly.

As part of the module generation tests, we create a .jl file for TEST_CAT.agf and include it. But now that the regular agf sources have already been built and included, we get an AGF GlassID clash. I either need to write some functionality for 

1. automatically appending AGF IDs instead of starting at 1, or
2. using something like GlassType.TEST to isolate this use case

tbh both would be useful. Just haven't gotten round to it yet as it's not important core functionality.